### PR TITLE
Teardown tables in reverse order for foreign keys

### PR DESCRIPTION
**Pull request by kumar303**
This is for PostgreSQL support which doesn't have a global
way to disable all foreign key constraints.

### CODE REVIEW

**Comment by kumar303** [comment]
@erikrose do you have a sec to look at this? I needed this to get things working in postgres. Tests are still really fast.

**Comment by erikrose** [comment]
I'll put it in my high-priority heap of django-nose patches to review this week[end]. I really with the test harness for django-nose itself was in better shape; not having a good way to test these patches really slows down the review.

**Comment by kumar303** [comment]
Yeah, testing test plugins can make the brain hurt.

File: test_utils/fixture_tables.py [hunk 1]
```diff
@@ -152,4 +152,6 @@ def read(self):
                     # fixture_name, humanize(fixture_dir)))
                     pass
 
-    return tables
+    # Return tables in reverse so that fixture deleting works as expected
+    # if you have foreign keys enabled (like for PostgreSQL).
+    return tables.reverse()
```
Comment (peterbe) [inline test_utils/fixture_tables.py:157]: That returns `None`
Surely you mean:

``` python
tables.reverse()
return tables
```

Or perhaps: `return reversed(tables)` but, mind you, that returns an iterator and not a list.

Comment (kumar303) [inline test_utils/fixture_tables.py:157]: well, huh. Good catch. Crazy! Apparently returning None fixed the tracebacks I was seeing in postgres. Now when I return the reversed tables I get another integrity error. I'll have to revisit this...

Comment (kumar303) [inline test_utils/fixture_tables.py:157]: For reference, this is what the traceback looks like (for a hobby site of mine):

```
Traceback (most recent call last):
  File "/Users/kumar/dev/recordbox/vendor/lib/python/nose/suite.py", line 227, in run
    self.tearDown()
  File "/Users/kumar/dev/recordbox/vendor/lib/python/nose/suite.py", line 350, in tearDown
    self.teardownContext(ancestor)
  File "/Users/kumar/dev/recordbox/vendor/lib/python/nose/suite.py", line 366, in teardownContext
    try_run(context, names)
  File "/Users/kumar/dev/recordbox/vendor/lib/python/nose/util.py", line 478, in try_run
    return func()
  File "/Users/kumar/dev/recordbox/vendor-local/test-utils-001f212/test_utils/__init__.py", line 159, in tearDownClass
    cls._fixture_teardown()
  File "/Users/kumar/dev/recordbox/vendor-local/test-utils-001f212/test_utils/__init__.py", line 214, in _fixture_teardown
    transaction.commit(using=db)
  File "/Users/kumar/dev/recordbox/vendor-local/django/db/transaction.py", line 138, in commit
    connection.commit()
  File "/Users/kumar/dev/recordbox/vendor-local/django/db/backends/__init__.py", line 223, in commit
    self._commit()
  File "/Users/kumar/dev/recordbox/vendor-local/django/db/backends/postgresql_psycopg2/base.py", line 235, in _commit
    return self.connection.commit()
IntegrityError: update or delete on table "keyword" violates foreign key constraint "keyword_id_refs_id_401b1e47" on table "x_ebayauction_keyword"
DETAIL:  Key (id)=(92960) is still referenced from table "x_ebayauction_keyword".
```

**Comment by kumar303** [comment]
ooh, github has a shiny new pull request dashboard and I need to clean out some ghosts :)

**Pull request closed without merging**